action/src/plugins/imageDiff/plugin.ts

```diff
@@ -0,0 +1,80 @@
+import * as core from '@actions/core';
+import { randomUUID } from 'crypto';
+import fs from 'fs';
+import download from 'image-downloader';
+import { basename } from 'path';
+import pixelmatch from 'pixelmatch';
+import { PNG } from 'pngjs';
+
+import { ImageDiffPlugin } from '../../schemas/stoatConfigSchema';
+import { GithubActionRun, UploadImageDiffRequest } from '../../types';
+import { submitPartialConfig } from '../helpers';
+import { createSignedUrl, uploadPath } from '../staticHosting/helpers';
+
+const runImageDiffPlugin = async (
+  taskId: string,
+  taskConfig: ImageDiffPlugin,
+  { ghToken, ghRepository: { repo, owner }, ghSha }: GithubActionRun,
+  stoatConfigFileId: number
+) => {
+  core.info(`[${taskId}] Running static hosting plugin (stoat config ${stoatConfigFileId})`);
+  const currentDirectory = process.cwd();
+  core.info(`[${taskId}] Current directory: ${currentDirectory}`);
+
+  // get baseline image
+  if (!taskConfig.baseline) {
+    throw new Error(`[${taskId}] Missing baseline image URL in ${taskId} task config`);
+  }
+  const uuid = randomUUID();
+  const baselinePath = `${currentDirectory}/${uuid}-baseline.png`;
+  core.info(`[${taskId}] Downloading baseline image to ${baselinePath}...`);
+  const baselineImageFile = await download.image({
+    url: taskConfig.baseline,
+    dest: baselinePath
+  });
+
+  // create diff image
+  const diffPath = `${currentDirectory}/${uuid}-diff.png`;
+  core.info(`[${taskId}] Creating image diff to ${diffPath}...`);
+  const baselineImage = PNG.sync.read(fs.readFileSync(baselinePath));
+  const currentImage = PNG.sync.read(fs.readFileSync(taskConfig.path));
```

Comment:
Can this use https://www.npmjs.com/package/jimp or similar to convert to PNG first so we can support jpeg/png/bmp/tiff/gif out of the box?

Reply:
Wow, this is super nice. Also this library can resize the image so that we can compare images with different sizes.

Reply:
Done. A downside of introducing `jimp` though is that this library is huge, 3MB, adding 88K lines of code to `dist/index.js`.